Applications/Spire/Source/Ui/ComboBox.cpp

```diff
@@ -125,7 +125,6 @@ bool ComboBox::eventFilter(QObject* watched, QEvent* event) {
         } else {
           revert_to(m_submission_text, false);
         }
-        return true;
```

Comment:
Briefly explain what this fixes.

Reply:
When the `ComboBox` is placed in the `OverlayPanel`, the Escape event is prevented from being sent to the `OverlayPanel` because of returning true after handling the Escape event in `ComboBox::eventFilter`, so the `OverlayPanel` can’t receive the Escape event to close itself.

Reply:
What alternative approach can you think of to resolve this issue?

Reply:
An alternative is that the `SecurityBox` listens to the Escape key press of the `TextBox` through `eventFilter`, when the `SecurityBox` receives the event, it resends the event to the `OverlayPanel`. It’s feasible because the `SecurityBox` receives the press event prior to the `ComboBox`.